src/services/user.service.js

```
@@ -0,0 +1,64 @@
+import { User } from '../data/user.js';
+import { Room } from '../data/room.js';
+import { roomService } from './room.service.js';
+
+const createUser = async (name) => {
+  const newUser = await User.create({
+    name,
+  });
+
+  return newUser;
+};
+
+const getUserById = async (id) => {
+  const user = await User.findOne({ where: { id } });
+
+  return user;
+};
+
+const getUserByName = async (name) => {
+  const user = await User.findOne({ where: { name } });
+
+  return user;
+};
+
+const findUsersByRoomId = async (roomId) => {
+  const room = await Room.findByPk(roomId, {
+    include: User,
+  });
+
+  if (!room) {
+    return [];
+  }
+
+  return room.Users;
+};
+
+const mergeUsers = async (roomId, targetRoomId) => {
+  const room = await roomService.findRoomById(roomId);
+  const targetRoom = await roomService.findRoomById(targetRoomId);
+  const usersToMerge = await findUsersByRoomId(roomId);
+  const targetUsers = await findUsersByRoomId(targetRoomId);
+
+  if (!room || !targetRoom) {
+    throw new Error('One or both rooms not found');
+  }
```

You fetch `usersToMerge` and `targetUsers` before validating `room` and `targetRoom`. If either room does not exist you’ll have performed unnecessary DB queries. Move the existence check (`if (!room || !targetRoom)`) to immediately after obtaining `room` and `targetRoom` (i.e. right after lines 38–39) so you fail fast and avoid extra work.